vendor/diff-parsers-ext-parsewyc.patch

```diff
@@ -0,0 +1,978 @@
++      { unclosed "{" $loc($1) "}" $loc($3) }*)
+   | od=open_dot_declaration DOT LBRACE record_expr_content RBRACE
+       { let (exten, fields) = $4 in
+         Pexp_open(od, mkexp ~loc:($startpos($3), $endpos)
+                         (Pexp_record(fields, exten))) }
+-  | mod_longident DOT LBRACE record_expr_content error
+-      { unclosed "{" $loc($3) "}" $loc($5) }
++  (*| mod_longident DOT LBRACE record_expr_content error
++      { unclosed "{" $loc($3) "}" $loc($5) }*)
+   | LBRACKETBAR expr_semi_list BARRBRACKET
+       { Pexp_array($2) }
+-  | LBRACKETBAR expr_semi_list error
+-      { unclosed "[|" $loc($1) "|]" $loc($3) }
++  (*| LBRACKETBAR expr_semi_list error
++      { unclosed "[|" $loc($1) "|]" $loc($3) }*)
+   | LBRACKETBAR BARRBRACKET
+       { Pexp_array [] }
+   | od=open_dot_declaration DOT LBRACKETBAR expr_semi_list BARRBRACKET
+       { Pexp_open(od, mkexp ~loc:($startpos($3), $endpos) (Pexp_array($4))) }
+   | od=open_dot_declaration DOT LBRACKETBAR BARRBRACKET
+       { (* TODO: review the location of Pexp_array *)
+         Pexp_open(od, mkexp ~loc:($startpos($3), $endpos) (Pexp_array [])) }
+-  | mod_longident DOT
++  (*| mod_longident DOT
+     LBRACKETBAR expr_semi_list error
+-      { unclosed "[|" $loc($3) "|]" $loc($5) }
++      { unclosed "[|" $loc($3) "|]" $loc($5) }*)
+   | LBRACKET expr_semi_list RBRACKET
+       { Pexp_list $2 }
+-  | LBRACKET expr_semi_list error
+-      { unclosed "[" $loc($1) "]" $loc($3) }
++  (*| LBRACKET expr_semi_list error
++      { unclosed "[" $loc($1) "]" $loc($3) }*)
+   | od=open_dot_declaration DOT LBRACKET expr_semi_list RBRACKET
+       { let list_exp = mkexp ~loc:($startpos($3), $endpos) (Pexp_list $4) in
+         Pexp_open(od, list_exp) }
+   | od=open_dot_declaration DOT mkrhs(LBRACKET RBRACKET {Lident "[]"})
+       { Pexp_open(od, mkexp ~loc:$loc($3) (Pexp_construct($3, None))) }
+-  | mod_longident DOT
++  (*| mod_longident DOT
+     LBRACKET expr_semi_list error
+-      { unclosed "[" $loc($3) "]" $loc($5) }
++      { unclosed "[" $loc($3) "]" $loc($5) }*)
+   | od=open_dot_declaration DOT LPAREN MODULE ext_attributes module_expr COLON
+     package_type RPAREN
+       { let modexp =
+           mkexp_attrs ~loc:($startpos($3), $endpos)
+             (Pexp_constraint (ghexp ~loc:$sloc (Pexp_pack $6), $8)) $5 in
+         Pexp_open(od, modexp) }
+-  | mod_longident DOT
++  (*| mod_longident DOT
+     LPAREN MODULE ext_attributes module_expr COLON error
+-      { unclosed "(" $loc($3) ")" $loc($8) }
++      { unclosed "(" $loc($3) ")" $loc($8) }*)
+ ;
+ labeled_simple_expr:
+     simple_expr %prec below_HASH
+       { (Nolabel, $1) }
+   | LABEL simple_expr %prec below_HASH
+@@ -2495,12 +2490,12 @@
+ ;
+ let_binding_body:
+   | let_binding_body_no_punning
+       { let p,e = $1 in (p,e,false) }
+ /* BEGIN AVOID */
+-  | val_ident %prec below_HASH
+-      { (mkpatvar ~loc:$loc $1, mkexpvar ~loc:$loc $1, true) }
++  (*| val_ident %prec below_HASH
```

Comment:
why is this commented

Reply:
All the rules between `/* BEGIN AVOID */ ... /* END AVOID */` (I didn't comment the full declaration between comments as it does not matter as long as I comment the calls) should be commented for `parse-wyc` otherwise the error rules of the parser are triggered instead of the recovery mode. If I uncomment this one I get regressions with parse-wyc.